src/Microsoft.Framework.Caching.SqlServer/SqlQueries.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Caching.SqlServer
+{
+    internal class SqlQueries
+    {
+        private const string CreateTableFormat = "CREATE TABLE {0}(" +
+            // add collation to the key column to make it case-sensitive
+            "Id nvarchar(100) COLLATE SQL_Latin1_General_CP1_CS_AS NOT NULL PRIMARY KEY, " +
+            "Value varbinary(MAX) NOT NULL, " +
+            "ExpiresAtTimeUTC datetime NOT NULL, " +
+            "SlidingExpirationInTicks bigint NULL," +
+            "AbsoluteExpirationUTC datetime NULL)";
+
+        private const string CreateNonClusteredIndexOnExpirationTimeFormat
+            = "CREATE NONCLUSTERED INDEX Index_ExpiresAtTimeUTC ON {0}(ExpiresAtTimeUTC)";
+
+        private const string TableInfoFormat =
+            "SELECT TABLE_CATALOG, TABLE_SCHEMA, TABLE_NAME, TABLE_TYPE " +
+            "FROM INFORMATION_SCHEMA.TABLES " +
+            "WHERE TABLE_SCHEMA = '{0}' " +
+            "AND TABLE_NAME = '{1}'";
+
+        private const string GetCacheItemFormat =
+            "SELECT Id, ExpiresAtTimeUTC, SlidingExpirationInTicks, AbsoluteExpirationUTC, Value " +
+            "FROM {0} WHERE Id = @Id AND @UtcNow <= ExpiresAtTimeUTC";
+
+        private const string GetCacheItemExpirationInfoFormat =
+            "SELECT Id, ExpiresAtTimeUTC, SlidingExpirationInTicks, AbsoluteExpirationUTC " +
+            "FROM {0} WHERE Id = @Id AND @UtcNow <= ExpiresAtTimeUTC";
+
+        private const string SetCacheItemFormat =
+            "IF NOT EXISTS(SELECT Id FROM {0} WHERE Id = @Id) " +
+            "BEGIN " +
+                "INSERT INTO {0} " +
+                    "(Id, Value, ExpiresAtTimeUTC, SlidingExpirationInTicks, AbsoluteExpirationUTC) " +
+                    "VALUES (@Id, @Value, @ExpiresAtTimeUTC, @SlidingExpirationInTicks, @AbsoluteExpirationUTC) " +
+            "END " +
+            "ELSE " +
+            "BEGIN " +
+                "UPDATE {0} SET Value = @Value, ExpiresAtTimeUTC = @ExpiresAtTimeUTC " +
```

Comment:
You should set all fields, they can all change if I call Set twice with different options.